libqtile/widget/tasklist.py

```diff
@@ -238,7 +254,12 @@ def get_taskname(self, window):
 
     @property
     def windows(self):
-        return self.bar.screen.group.windows
+        if self.current_win_only:
+            wins = self.bar.screen.group.current_window
+            wins = set([wins]) if wins else set([])  # Sets are necessary here to ensure consistent formatting
+        else:
+            wins = self.bar.screen.group.windows.copy()  # set2 = set1 uses copy by reference
```

Comment:
Why does this need to be copied?

Reply:
The original grabs the set by reference so when tasklist.skip_taskbar edits the set it edits the object in memory that all of Qtile uses if we don't copy it. This causes the window to enter a half managed half not state if it has the skiptaskbar property.

Reply:
Interesting. I assumed `skip_taskbar` was creating a new set rather than editing the existing one as it's not changing `windows` but making a new set from items in `windows`.

Reply:
You're right. I forgot about that when I switched to set comprehension from an earlier clunkier version that modified the set in place.